src/checkers/inference/solver/frontend/LatticeBuilder.java

```diff
@@ -0,0 +1,194 @@
+package checkers.inference.solver.frontend;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+public class LatticeBuilder {
+
+    /**
+     * subType maps each type qualifier to its sub types.
+     */
```

Comment:
Since the outside callers will use the `Lattice` instead of `LatticeBuilder`, it would be better to move those javadocs to `Lattice`, otherwise developers cannot see those javadocs when they use `Lattice`.